cmd/datamon/cmd/bundle_upload.go

```diff
@@ -63,9 +65,31 @@ var uploadBundleCmd = &cobra.Command{
 			core.BlobStore(blobStore),
 			core.ConsumableStore(sourceStore),
 			core.MetaStore(MetaStore),
+			core.SkipMissing(bundleOptions.SkipOnError),
 		)
 
-		err = core.Upload(context.Background(), bundle)
+		var fn func() ([]string, error)
+
+		if bundleOptions.FileList != "" {
+			fn = func() ([]string, error) {
+				file, err := os.Open(bundleOptions.FileList)
+				if err != nil {
+					return nil, fmt.Errorf("failed to open file: %s err:%s", bundleOptions.FileList, err.Error())
+				}
+				lineScanner := bufio.NewScanner(file)
+				files := make([]string, 0)
+				for lineScanner.Scan() {
+					files = append(files, lineScanner.Text())
+				}
+				return files, nil
+			}
+		} else {
+			fn = func() ([]string, error) {
```

Comment:
note that function types can be nil, so it's possible to allow `core/` to specify the default behavior on `nil`